fields/sql.go

```diff
@@ -0,0 +1,106 @@
+package fields
+
+import (
+	"database/sql/driver"
+	"encoding"
+	"encoding/json"
+	"reflect"
+)
+
+// Valuer fulfills the sql/driver.Valuer interface which deserializes our
+// struct field value into a valid SQL value.
+type Valuer struct {
+	*Descriptor
+	value reflect.Value
+}
+
+// Value satisfies the sql/driver.Valuer interface.
+// The value should be one of the following:
+//    int64
+//    float64
+//    bool
+//    []byte
+//    string
+//    time.Time
+//    nil - for NULL values
+func (f Valuer) Value() (driver.Value, error) {
+	// Ideally, all the interfaces below would be implemented as normal methods, not indirect
+	// methods. However, in order to be safe we take the address of pointers where we can.
+	// This specifically supports protobuf.
```

Comment:
is this comment referring to the next line? i am having trouble understanding what this implies for my read here